backend/testing/tester.py

```diff
@@ -0,0 +1,216 @@
+
+    def test(self):
+        '''
+        Loads a test file that includes crafted bgp updates as input and expected messages as output.
+        '''
+
+        RABBITMQ_HOST = os.getenv('RABBITMQ_HOST')
+
+        # exchanges
+        update_exchange = Exchange(
+            'bgp-update',
+            type='direct',
+            durable=False,
+            delivery_mode=1)
+
+        hijack_exchange = Exchange(
+            'hijack-update',
+            type='direct',
+            durable=False,
+            delivery_mode=1)
+
+        pg_amq_bridge = Exchange(
+            'amq.direct',
+            type='direct',
+            durable=True,
+            delivery_mode=1)
+
+        # queues
+        update_queue = Queue(
+            'detection-testing',
+            exchange=pg_amq_bridge,
+            routing_key='update-insert',
+            durable=False,
+            auto_delete=True,
+            max_priority=1,
+            consumer_arguments={'x-priority': 1})
+
+        hijack_queue = Queue(
+            'hijack-testing',
+            exchange=hijack_exchange,
+            routing_key='update',
+            durable=False,
+            auto_delete=True,
+            max_priority=1,
+            consumer_arguments={'x-priority': 1})
+
+        hijack_db_queue = Queue(
+            'hijack-db-testing',
+            exchange=pg_amq_bridge,
+            routing_key='hijack-update',
+            durable=False,
+            auto_delete=True,
+            max_priority=1,
+            consumer_arguments={'x-priority': 1})
+
+        messages = {}
+        # load test
+        with open('messages.json', 'r') as f:
+            messages = json.load(f)
+
+        send_len = len(messages)
+
+        def validate_message(body, message):
+            '''
+            Callback method for message validation from the queues.
+            '''
+            print('\t- Receiving Batch #{} - Type {} - Remaining {}'.format(self.curr_idx,
+                                                                            message.delivery_info['routing_key'], self.expected_messages - 1))
+            if isinstance(body, dict):
+                event = body
+            else:
+                event = json.loads(body)
+            # logging.debug(event)
+
+            # distinguish between type of messages
+            if message.delivery_info['routing_key'] == 'update-insert':
+                expected = messages[self.curr_idx]['detection_update_response']
+            elif message.delivery_info['routing_key'] == 'update':
+                expected = messages[self.curr_idx]['detection_hijack_response']
+            elif message.delivery_info['routing_key'] == 'hijack-update':
+                expected = messages[self.curr_idx]['database_hijack_response']
+
+            # compare expected message with received one. exit on mismatch.
+            for key in set(event.keys()).intersection(expected.keys()):
+                if event[key] != expected[key]:
```

Comment:
what happens with the ordering of e.g., lists like asns_inf? Shouldn't these be compared in terms of content (e.g., if set(asns_inf_A) == set(asns_inf_B))? Just wondering if we get sth wrong in the expected order, the test will fail. If we require this by design, you can resolve this comment.

Reply:
we make it set comparison as order does not matter